### PR TITLE
fix: Creation of ingress resource with k8s trigger

### PR DESCRIPTION
**Pull request by daniel-codefresh**
Signed-off-by: Daniel Soifer <daniel.soifer@codefresh.io>

Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

### CODE REVIEW

**Comment by daniel-codefresh** [comment]
Fixes: #1432

File: sensors/triggers/triggers.go [hunk 1]
```diff
@@ -26,9 +25,17 @@ import (
 
 func GetGroupVersionResource(obj *unstructured.Unstructured) schema.GroupVersionResource {
 	gvk := obj.GroupVersionKind()
+	lowercaseKind := strings.ToLower(gvk.Kind)
```
Comment (whynowy) [inline sensors/triggers/triggers.go:28]: Can we use this?

https://github.com/kubernetes/gengo/blob/master/namer/plural_namer.go#L54

Comment (daniel-codefresh) [reply sensors/triggers/triggers.go:28]: Yes, good point, changing now

**Review by whynowy — APPROVED** [review]
Excellent!

**Comment by daniel-codefresh** [comment]
@whynowy BTW, I just noticed we are using `eventbus` as the plural form of the EventBus CRD, is there any particular reason for this? It got me thinking that maybe we should have an optional `resource` field after all
https://github.com/argoproj/argo-events/blob/7ad602783e590898909ac4c6682c5a66c6b21f0e/hack/crds.go#L32

**Comment by whynowy** [comment]
> @whynowy BTW, I just noticed we are using `eventbus` as the plural form of the EventBus CRD, is there any particular reason for this? It got me thinking that maybe we should have an optional `resource` field after all
> 
> https://github.com/argoproj/argo-events/blob/7ad602783e590898909ac4c6682c5a66c6b21f0e/hack/crds.go#L32

Per @alexec (who is from England) - plural of `bus` is still `bus`...

**Comment by daniel-codefresh** [comment]
@whynowy Oh I see.. Well I'm certainly not an English expert but from a quick Google search, it seems like `buses` is also an acceptable plural form of `bus`:
https://www.oxfordlearnersdictionaries.com/definition/english/bus_1
https://www.collinsdictionary.com/dictionary/english/bus
https://www.merriam-webster.com/words-at-play/plural-of-bus
And more importantly, I just verified that the `k8s.io/gengo/namer` library that was used in this PR is actually returning `eventbuses` as the plural form of `eventbus` (which will cause a failure if a user tries to CRUD an `eventbus` through a k8s-trigger), so I guess we would have to support that by one way or another... WDYT?

**Comment by whynowy** [comment]
> @whynowy Oh I see.. Well I'm certainly not an English expert but from a quick Google search, it seems like `buses` is also an acceptable plural form of `bus`: https://www.oxfordlearnersdictionaries.com/definition/english/bus_1 https://www.collinsdictionary.com/dictionary/english/bus https://www.merriam-webster.com/words-at-play/plural-of-bus And more importantly, I just verified that the `k8s.io/gengo/namer` library that was used in this PR is actually returning `eventbuses` as the plural form of `eventbus` (which will cause a failure if a user tries to CRUD an `eventbus` through a k8s-trigger), so I guess we would have to support that by one way or another... WDYT?

I think it would be weird to support both - an extra CRD for `buses`. I would recommend to keep it as is. If there's really somebody want to use that to CRUD an `eventbus`, we should add an exception when using that lib. I remember `code-nenerator` also used to support plural `exceptions`, now sure why it stopped working, and I added that workaround in the shell.

**Comment by daniel-codefresh** [comment]
@whynowy Yes, definitely, an extra CRD would be weird. I actually meant either adding an exception like you suggested, or changing the plural form of the existing CRD back to `eventbuses` so we can stick to the conventions and also get rid of that extra logic that was added to the codegen process, of transforming `eventbuses` to `eventbus`

**Comment by daniel-codefresh** [comment]
> > @whynowy Oh I see.. Well I'm certainly not an English expert but from a quick Google search, it seems like `buses` is also an acceptable plural form of `bus`: https://www.oxfordlearnersdictionaries.com/definition/english/bus_1 https://www.collinsdictionary.com/dictionary/english/bus https://www.merriam-webster.com/words-at-play/plural-of-bus And more importantly, I just verified that the `k8s.io/gengo/namer` library that was used in this PR is actually returning `eventbuses` as the plural form of `eventbus` (which will cause a failure if a user tries to CRUD an `eventbus` through a k8s-trigger), so I guess we would have to support that by one way or another... WDYT?
> 
> I think it would be weird to support both - an extra CRD for `buses`. I would recommend to keep it as is. If there's really somebody want to use that to CRUD an `eventbus`, we should add an exception when using that lib. I remember `code-nenerator` used to support plural `exceptions`, now sure why it stopped working, and I added that workaround in the shell.

@whynowy Yes, definitely, an extra CRD would be weird. I actually meant either adding an exception like you suggested, or changing the plural form of the existing CRD back to eventbuses so we can stick to the conventions and also get rid of that extra logic that was added to the codegen process, of transforming eventbuses to eventbus

**Comment by whynowy** [comment]
@daniel-codefresh - I prefer to add exceptions, changing CRD definition sometimes brings weird issues during upgrade. Code-gen used to support [exceptions](https://github.com/argoproj/argo-events/blob/v1.3.1/hack/update-codegen.sh#L41), not sure why it is not supporting now. I didn't spent time on the research.

**Comment by daniel-codefresh** [comment]
> @daniel-codefresh - I prefer to add exceptions, changing CRD definition sometimes brings weird issues during upgrade. Code-gen used to support [exceptions](https://github.com/argoproj/argo-events/blob/v1.3.1/hack/update-codegen.sh#L41), not sure why it is not supporting now. I didn't spent time on the research.

@whynowy Oh yea, now I see what you meant by codegen exceptions.. I was thinking about using a plural exception for `eventbus` together with `k8s.io/gengo/namer ` lib so we can be covered in case a user wants to CRUD eventbus resource as well. Heres the change I was talking about: #1440

**Pull request merged**